tests/cms/views/pages/test_page_actions.py

```diff
@@ -0,0 +1,228 @@
+    child_lft = start_lft + 1
+    for i in range(num_children):
+        Page.objects.create(
+            region=root_page.region,
+            parent=root_page,
+            lft=child_lft,
+            rgt=child_lft + 1,
+            tree_id=tree_id,
+            depth=depth + 1,
+        )
+        PageTranslation.objects.create(
+            page=root_page.children.last(),
+            language=root_page.region.default_language,
+            title=f"Child {i}{root_name_add}",
+            slug=f"child-{i}{root_name_add}",
+            status=status.PUBLIC,
+        )
+        child_lft += 2
+
+    return root_page.id, root_rgt
+
+
+def create_page_with_mirror(
+    region_slug: str,
+    name_add: str = "",
+    start_lft: int = 1,
+    tree_id: int = 1,
+) -> tuple[int, int]:
+    """
+    A helper function to create a test page with a mirror page
+
+    Returns the page ID and the mirrored page ID
+    """
+    region = Region.objects.get(slug=region_slug)
+    original_page = Page.objects.get(
+        id=create_page(
+            region_slug,
+            name_add=f"{name_add}-original",
+            lft=start_lft,
+            rgt=start_lft + 1,
+            tree_id=tree_id,
+            depth=1,
+        )
+    )
+
+    mirrored_page = Page.objects.create(
+        region=region,
+        mirrored_page=original_page,
+        mirrored_page_first=True,
+        tree_id=tree_id + 1,
+        lft=start_lft,
+        rgt=start_lft + 1,
+        depth=1,
+    )
+
+    PageTranslation.objects.create(
+        page=mirrored_page,
+        language=region.default_language,
+        title="Mirrored Page" + name_add,
+        slug="mirrored-page" + name_add,
+        status=status.PUBLIC,
+    )
+    return original_page.id, mirrored_page.id
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize("role", ["ROOT", "AUTHOR"])
+@pytest.mark.parametrize(
+    "num_deletable, num_undeletable_1, num_undeletable_2",
+    [
+        pytest.param(
+            1, 1, 1, id="deletable_page=1-page_with_children=1-page_with_mirror=1"
+        ),
+        pytest.param(2, 0, 0, id="deletable_pages=2"),
+        pytest.param(0, 2, 2, id="pages_with_children=2-pages_with_mirror=2"),
+        pytest.param(
+            2, 2, 2, id="deletable_pages=2-pages_with_children=2-pages_with_mirror=2"
+        ),
+    ],
+)
+def test_bulk_delete_pages(
+    role: str,
+    client: Client,
+    load_test_data: None,
+    settings: SettingsWrapper,
+    caplog: LogCaptureFixture,
+    num_deletable: int,
+    num_undeletable_1: int,
+    num_undeletable_2: int,
+) -> None:
+    """
+    Test whether bulk deleting of pois works as expected
+    """
+    user = get_user_model().objects.get(username=role.lower())
+    client.force_login(user)
+
+    deletable_pages = [
+        create_page(
+            "augsburg",
+            lft=2 * i + 1,
+            rgt=2 * i + 2,
+            depth=1,
+            tree_id=20,
+            name_add=f"-{i}",
+        )
```

Comment:
Why exactly are you specifying the exact `lft`, `rgt`, `depth` and `tree_id` values again instead of using treebeards functions? If this is out of them being suspicious at first and you wanting to be explicit with what will end up in the database, I feel you. But I think this is effort spent in the wrong place – as soon as our test data grows to 20 tree structures, this will produce a collision with manually setting the `tree_id`. But we're not even testing at the level of values internal to treebeard here, just the pages involved being in the correct parent-child relationship with each other is sufficient. So throughout these tests I'm second guessing this choice here